users/userList.js

```diff
@@ -1,23 +1,41 @@
+const email = require("../helpers/email");
+
 module.exports = function makeUserList(db_model) {
+    return Object.freeze({
+        add,
+        remove,
+        findOneByEmail,
+        updateEmail,
+        updateFirstName,
+        updateLastName,
+    });
+
+    async function add(user) {
+        return await db_model.create(user);
+    }
 
-	return Object.freeze({
-		add,
-		remove,
-		findOneByEmail,
-	});
+    async function remove(user = {}) {
+        const results = await db_model.deleteMany(user);
+        return results.deletedCount;
+    }
 
-	async function add(user) {
-		return await db_model.create(user);
-	}
+    async function findOneByEmail(email) {
+        return await db_model.findOne({ email: email });
+    }
 
-	async function remove(user = {}) {
-		const results = await db_model.deleteMany(user);
+    async function updateEmail(email, newEmail) {
+        const updatedUser = await db_model.findOneAndUpdate({email: email}, {email: newEmail}, {new: true})
+        return updatedUser;
+    }
 
-		return results.deletedCount;
-	}
+    async function updateFirstName(firstName, newFirstName) {
```

Comment:
It seems weird that we are searching for the user by first name. What if there are several users with the same first name? (Same with last name below)

Reply:
This is some old tests, and i don't think that we even use the functions in the project at all, maybe just delete them